Orochi/OrochiUtils.h

```diff
@@ -19,7 +19,10 @@ class OrochiUtils
 	};
 
 	static oroFunction getFunctionFromFile( oroDevice device, const char* path, const char* funcName, std::vector<const char*>* opts );
-	static oroFunction getFunction( oroDevice device, const char* code, const char* path, const char* funcName, std::vector<const char*>* opts );
+	static oroFunction getFunctionFromString( oroDevice device, const char* source, const char* path, const char* funcName, std::vector<const char*>* opts, 
```

Comment:
Since you are modifying the function prototype of these utilities, I would like to suggest to change `const char*`  to `const std::string&` since it's safer. (`const char*` has no information about what the size is and does not guarantee if the underlying memory space exists, nor would it guarantee it will be cleaned up properly).

Another observation is that, inside body of the `getFunction`  we do use `std::string` to catch and process some of the parameters.  Some of these parameters are solely inputs and are not meant to be changed so they can be constant references . 

```suggestion
	static oroFunction getFunctionFromString( oroDevice device, const std::string& source, const std::string& path, const std::string& funcName, const std::vector<std::string>& opts, 
```

Just a suggestion, please feel free to accept or decline 😃

Reply:
postponed for later